### PR TITLE
feat: map nostr public key to DNS joyboy.community using NIP-05

### PR DESCRIPTION
**Pull request by Gift-Naomi**

### CODE REVIEW

File: website/src/app/api/.well-known/nostr.json/route.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const nostr_keys = {
+  joyboy: {
+    pubkey: 'npub1nq5ckz62p4vxwu08lpx8ggu5k5qn6d7pdtcfyj7hae3wc6j30fwseex2rq',
+  },
+  testnet: {
+    pubkey: 'b0635d6a9851d3aed0cd6c495b282167acf761729078d975fc341b22650b07b9'
+  }
+};
+
+export async function GET(request: NextRequest) {
```
Comment (maciejka) [inline website/src/app/api/.well-known/nostr.json/route.ts:12]: Will this be available at `/.well-known/nostr.json` or `/api/.well-known/nostr.json`?

File: website/src/app/page.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { ContributeSection } from "./components/ContributeSection";
+import { DescriptionSection } from "./components/DescriptionSection";
+import { DownloadSection } from "./components/DownloadSection";
+import { Faq } from "./components/Faq";
+import { Footer } from "./components/Footer";
+import { HeroSection } from "./components/HeroSection";
+import { Navbar } from "./components/Navbar";
+
+export default function App() {
+  return (
```
Comment (maciejka) [inline website/src/app/page.tsx:11]: Not sure if this is the right file, this app should return current content from: https://www.joyboy.community/.

**Review by maciejka — CHANGES_REQUESTED** [review]

**Comment by drspacemn** [comment]
Screenshot from 2024-05-24 10-10-08

We should be able to delete the vercel project and reinit it as a next.js app

File: website/src/app/api/.well-known/nostr.json/route.ts (hunk 1 above)
Comment (Gift-Naomi) [reply website/src/app/api/.well-known/nostr.json/route.ts:12]: oh yeah it's suppose to be available at ```/.well-known/nostr.json```. I'll refactor it

File: website/src/app/page.tsx (hunk 2 above)
Comment (Gift-Naomi) [reply website/src/app/page.tsx:11]: Yea, it does return the current content from  https://www.joyboy.community/.

**Review by maciejka — COMMENTED** [review]
Hey @Gift-Naomi, would you please add a screenshot in the comment to: https://github.com/keep-starknet-strange/joyboy/pull/95 showing that /.well-known/nostr.json works in your local environment?

**Comment by Gift-Naomi** [comment]
@maciejka this is the image below

image

**Comment by maciejka** [comment]
@drspacemn looks like we can recreate this as next.js app in vercel.

**Comment by maciejka** [comment]
> @maciejka please find the screenshot below
> ...

Thanks! Please resolve conflicts.

**Comment by Gift-Naomi** [comment]
> > @maciejka please find the screenshot below
> > ...
> 
> Thanks! Please resolve conflicts.

Resolved!

**Comment by maciejka** [comment]
> > > @maciejka please find the screenshot below
> > > ...
> > 
> > 
> > Thanks! Please resolve conflicts.
> 
> Resolved!

Thanks! Let's now wait for @drspacemn to fix the deployment.

**Comment by maciejka** [comment]
@Gift-Naomi I consider this pr as complete. We just need to wait for vercel reset.

**Comment by Gift-Naomi** [comment]
> @Gift-Naomi I consider this pr as complete. We just need to wait for vercel reset.

Alright.

**Comment by maciejka** [comment]
@Gift-Naomi would you please resolve conflicts?
@drspacemn will you have a look at vercel deployment?

**Comment by Gift-Naomi** [comment]
> @Gift-Naomi would you please resolve conflicts? @drspacemn will you have a look at vercel deployment?

Okay

**Comment by Gift-Naomi** [comment]
> @Gift-Naomi would you please resolve conflicts? @drspacemn will you have a look at vercel deployment?

Done!

**Comment by drspacemn** [comment]
So to confirm you want me to re-initialize this project with the website subdir as the root, correct?
Keep in mind this won't build the branch from the PR only main

**Review by maciejka — APPROVED** [review]

**Review by maciejka — APPROVED** [review]

**Pull request merged**